### PR TITLE
turns stickmen into basic mobs

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

turns stickmen into basic mobs
makes the paper wizard robe stickmen be hostile to people that arent stickmen and arent you 

## Why It's Good For The Game

more basic mobs

## Changelog

:cl:
refactor: turns stickmen into basic mobs
balance: wizard paper robe now makes stickmen hostile to things that arent stickmen and arent you
/:cl:

### CODE REVIEW

**Comment by maxymax13** [comment]
what the fuck are basic mobs? simple mobs were made to avoid having to simulate all the extra garbage but now they've become bloated themselves. soon in 8 years you'll have to make simpler mobs because basic mobs got bloated too

**Comment by Fikou** [comment]
> what the fuck are basic mobs? simple mobs were made to avoid having to simulate all the extra garbage but now they've become bloated themselves. soon in 8 years you'll have to make simpler mobs because basic mobs got bloated too

dumbass

**Comment by AMonkeyThatCodes** [comment]
what fikou said. Simple mobs were coded in the literal stone age when composition and datum AI didn't exist. You could say the same about any refactor because yes, someday in the future someone might have an idea how to improve basic mobs.

File: code/modules/mob/living/basic/vermin/stickman.dm [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		/datum/ai_planning_subtree/find_and_hunt_target
+	)
+
+/datum/ai_planning_subtree/basic_melee_attack_subtree/stickman
+	melee_attack_behavior = /datum/ai_behavior/basic_melee_attack/stickman
+
+/datum/ai_behavior/basic_melee_attack/stickman
+	action_cooldown = 1.5 SECONDS
+
+/datum/ai_controller/basic_controller/stickman/PerformIdleBehavior(delta_time)
```
Comment (AMonkeyThatCodes) [inline code/modules/mob/living/basic/vermin/stickman.dm:45]: Not a note for you necesarily but I'll make idle behaviors into a datum. this piece of code is copy pasted all over the place.

File: code/modules/mob/living/basic/vermin/stickman.dm [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	gender = MALE
+	health = 100
+	maxHealth = 100
+	speed = 0
+	attack_verb_continuous = "punches"
+	attack_verb_simple = "punch"
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	attack_sound = 'sound/weapons/punch1.ogg'
+	combat_mode = TRUE
```
Comment (AMonkeyThatCodes) [inline code/modules/mob/living/basic/vermin/stickman.dm:17]: i don't think this is needed (?) can u test?

Comment (Fikou) [reply code/modules/mob/living/basic/vermin/stickman.dm:17]: isnt this what prevents you from swapping places

File: code/modules/mob/living/basic/vermin/stickman.dm [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+/datum/ai_controller/basic_controller/stickman
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic()
+	)
+
+	ai_movement = /datum/ai_movement/basic_avoidance
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree/stickman,
+		/datum/ai_planning_subtree/find_and_hunt_target
```
Comment (AMonkeyThatCodes) [inline code/modules/mob/living/basic/vermin/stickman.dm:36]: remove this one. this makes them hunt ants.

File: code/modules/mob/living/basic/vermin/stickman.dm (hunk 2 above)
Comment (AMonkeyThatCodes) [reply code/modules/mob/living/basic/vermin/stickman.dm:17]: oh true lol

**Pull request merged**